src/openforms/formio/rendering/default.py

```diff
@@ -345,3 +350,60 @@ def is_visible(self) -> bool:
         registration data...
         """
         return False
+
+
+@register("partners")
+class PartnersNode(ComponentNode):
+    @property
+    def value(self) -> Any:
+        value = glom(self.step_data.data, self.path or self.key)
+        return value
+
+    @property
+    def display_value(self) -> str | Any:
+        # in export mode, expose the raw datatype
+        if self.mode == RenderModes.export:
+            return self.value
+
+        if isinstance(self.value, list):
+            return ""
+
+        return format_value(self.component, self.value, as_html=self.renderer.as_html)
+
+    def get_children(self) -> Iterator[ComponentNode]:
+        repeats = len(self.value) if self.value else 0
+        fields = [
+            ("bsn", _("BSN")),
+            ("initials", _("Initials")),
+            ("affixes", _("Affixes")),
+            ("lastName", _("Lastname")),
+            ("dateOfBirth", _("Date of birth")),
+        ]
+
+        for node_index in range(repeats):
+            yield ComponentNode.build_node(
+                step_data=self.step_data,
+                component={
+                    "type": "component_label",
+                    "key": "component_label",
+                    "label": f"Partner {node_index + 1}",
+                },
+                renderer=self.renderer,
+                depth=self.depth + 1,
+                path=f"{self.key}.{node_index}",
+                parent_node=self,
+            )
+
+            for key, label in fields:
+                yield ComponentNode.build_node(
+                    step_data=self.step_data,
+                    component={
+                        "type": key,
```

Comment:
careful here - this must be a valid/supported component type, so I think you'll have to add that to `fields`. Though I'd recommend just doing something like:

```py
components: list[Component] = [
    {
        "type": "bsn",
        "key": "bsn",
        "label": _("BSN"),
    },
    {
        "type": "textfield",
        "key": "initials",
        "label": _("Initials"),
    },
    ...
]
```
which gives a bit more type-safety too and keeps with the existing pattern of defining the formio structure in memory.